File: Atividade_6/C/ex05.c
#include <stdio.h>
#include <stdlib.h>
#include <float.h>

int main() {
    char mes_ano[10];
    double ipca, soma_ipca = 0.0;
    double menor_ipca = DBL_MAX;
    double maior_ipca = -DBL_MAX;
    char menor_mes_ano[10], maior_mes_ano[10];
    int total_meses = 0;

    while (1) {
        scanf("%s", mes_ano);
        if (mes_ano[0] == '*') {
            break;
        }
        scanf("%lf", &ipca);

        soma_ipca += ipca;
        total_meses++;

        if (ipca < menor_ipca) {
            menor_ipca = ipca;
            sprintf(menor_mes_ano, "%s", mes_ano);
            getchar();
        }

        if (ipca > maior_ipca) {
            maior_ipca = ipca;
            sprintf(maior_mes_ano, "%s", mes_ano);
            getchar();
        }
    }

    printf("Menor IPCA: %.2lf (%s)\n", menor_ipca, menor_mes_ano);
    printf("Maior IPCA: %.2lf (%s)\n", maior_ipca, maior_mes_ano);
    printf("Media do IPCA: %.2lf\n", soma_ipca / total_meses);

    return 0;
}

File: Atividade_6/Haskell/ex05.hs
import Text.Printf (printf)

main :: IO ()
main = do
    putStrLn "Digite o mês e ano (Ex: 01/2021) e IPCA:"
    loop "" (1/0) "" (-1/0) 0 0
  where
    loop menorMesAno menorIpca maiorMesAno maiorIpca somaIpca totalMeses = do
        entrada <- getLine
        if entrada == "*"
            then do
                let mediaIpca = if totalMeses == 0 then 0 else somaIpca / fromIntegral totalMeses
                printf "Menor IPCA: %.2f no mês/ano %s\n" menorIpca menorMesAno
                printf "Maior IPCA: %.2f no mês/ano %s\n" maiorIpca maiorMesAno
                printf "Média do IPCA: %.2f\n" mediaIpca
            else do
                let (mesAno, rest) = break (== ' ') entrada
                    ipcaStr = drop 1 rest
                    ipca = read ipcaStr :: Double
                loop (if ipca < menorIpca then mesAno else menorMesAno)
                     (min menorIpca ipca)
                     (if ipca > maiorIpca then mesAno else maiorMesAno)
                     (max maiorIpca ipca)
                     (somaIpca + ipca)
                     (totalMeses + 1)
